### ISSUE TITLE
Support hCaptcha

### ISSUE THREAD

**Issue by dessant** [dessant/buster#319]
Subscribe to this issue for updates on hCaptcha support.

**Comment by philibeur** [comment]
+1

**Comment by sonuyos** [comment]
Any hopes of this happening?

**Comment by Pandapip1** [comment]
This is not needed. hCaptcha supports a really good accessibility system that automatically bypasses captchas for you: https://www.hcaptcha.com/accessibility

**Comment by kotx** [comment]
That requires a signup/email, though.

**Comment by Dhanamerdeka** [comment]
I had some idea....
how about buster add some feature to automatically get the cookies method to bypass it?

example, you click this link https://accounts.hcaptcha.com/verify_email/72e6d546-af6e-4816-8c33-9d9cb91a6f00
and then you click set cookies....

after that most of hcaptcha will be bypassed....
can you automate that?
i think this will be a great idea.... sometime if we exit browser/use it next day... we should do this again and again

**Comment by Pandapip1** [comment]
That seems like a pretty good idea, if it can be pulled off. I'm hopeful it can be.

**Comment by Dhanamerdeka** [comment]
> i think this will be a great idea.... sometime if we exit browser/use it next day... we should do this again and again

yeah :D i hope @dessant can do that :) 

google captcha done...
hcaptcha now in the next :D

**Comment by Pandapip1** [comment]
Although, now that I think of it, that still requires an email.

**Comment by KaKi87** [comment]
> automatically get the cookies method to bypass it

I made [a PoC](https://git.kaki87.net/KaKi87/hcaptcha-accessibility-cookie/src/branch/master/index.js) for that once, although it must have broken by now, but I'm just saying it's doable. :+1:

**Comment by leodoi3** [comment]
I see there's already an extension for solving hCaptcha.  https://chrome.google.com/webstore/detail/hcaptcha-solver/lfpfbgeoodeejmjdlfjbfjkemjlblijg
Although they don't have their source available , my guess is image recogniziion, but it's hard to say how they do it.

**Comment by Dhanamerdeka** [comment]
> > automatically get the cookies method to bypass it
> 
> I made [a PoC](https://git.kaki87.net/KaKi87/hcaptcha-accessibility-cookie/src/branch/master/index.js) for that once, although it must have broken by now, but I'm just saying it's doable. 👍

why they had temp-mail?
didn't the method just use the get cookies?

**Comment by Dhanamerdeka** [comment]
> I see there's already an extension for solving hCaptcha. https://chrome.google.com/webstore/detail/hcaptcha-solver/lfpfbgeoodeejmjdlfjbfjkemjlblijg Although they don't have their source available , my guess is image recogniziion, but it's hard to say how they do it.

i see... i will check it...
but still i hope they will add it to buster... so we can had 1 extension for most major captcha

**Comment by KaKi87** [comment]
> why they had temp-mail?

AFAIK there is no way to get the cookie without receiving an email, so I used a disposable email service to receive it.

**Comment by Dhanamerdeka** [comment]
> > why they had temp-mail?
> 
> AFAIK there is no way to get the cookie without receiving an email, so I used a disposable email service to receive it.

i still didn't understand what you mean.....
i mean... the hcaptcha works when you click get cookies....

only that... no need for any new mail or anything....
i just didn't understand what you mean by receiving email....

we can just use 1 get cookies like what i shared above and all will works like a rock....
or maybe i miss somethhing lol XD

**Comment by KaKi87** [comment]
Wow, I didn't realized the cookie could be shared.

Well, hCaptcha will most likely find out some day and prevent accessibility cookies from being used on multiple devices.

**Comment by Pandapip1** [comment]
The accessibility cookie is (supposedly) rate limited. This will (likely) not work.

**Comment by Dhanamerdeka** [comment]
> Wow, I didn't realized the cookie could be shared.
> 
> Well, hCaptcha will most likely find out some day and prevent accessibility cookies from being used on multiple devices.

i think they not shared the cookie
but they recognize you.... or add some setting to make your devices skip captcha.....

maybe that's they do on practical use....
and they allow multiple device to use same link is to allow people who had multiple devices to be able to skip the link with 1 single account.....

maybe something like that

**Comment by kotx** [comment]
You need an email to get a link, which is rate-limited and therefore can't be shared. Automating a disposable mail address is how you get that link.

**Comment by KaKi87** [comment]
On the contrary, you can't make an API to scrape a rate-limited service, since all requests will originate from the server's static IP address, which will eventually be blocked.

Also, you can't constantly use the same disposable email provider, because it will get detected and eventually banned as well, considering that many of those services already are blocked.

You could submit the form client-side and receive the email server-side, but then the IP address will be blocked by the disposable email providers instead.

So, a fully client-side solution would be the best implementation of the email method, the downside is that you need an actual system app to run a headless browser capable of scraping both hCaptcha and email providers, which would communicate with the extension.

However, I would personally recommend looking for another method than the email one.

**Comment by Dhanamerdeka** [comment]
the most proven ways from me....

USE YOUR OWN EMAIL, and get the verification link from hcaptcha itself....
then atleast... use addons to automate "get cookies" every 4 hour....

i think that's simplest ways to do it.... no need disposable or anything....
just works like old busters....

on old ones... you need to place key.... and with hcaptcha... you just need to place the verification link, and let the addons do the job.....

i can't find any more simple answer for this.... maybe my idea can help

**Comment by KaKi87** [comment]
> USE YOUR OWN EMAIL [...] use addons to automate "get cookies" every 4 hour

I think automating personal e-mail addresses is more sensitive and trickier than disposable emails.

**Comment by Dhanamerdeka** [comment]
> > USE YOUR OWN EMAIL [...] use addons to automate "get cookies" every 4 hour
> 
> I think automating personal e-mail addresses is more sensitive and trickier than disposable emails.

hmmm maybe you didn't even know what i thinking and how hcaptcha works didn't u?
nothing will lose if you use new email, old email, personal email, or even your parent email....

you just need to copy THE LINK AND CLICK GET THE COOKIES AND DONE
no data leak... you safe... other's safe....

as simple as that....ah yeah... if you says it's sensitive... then the api key inside busters addons also sensitive... all data that we had include our real name is sensitive....

i didn't know what you mean above....and where the point is... don't make it difficult for yourself :(

**Comment by KaKi87** [comment]
The methods used to automatically receive and read emails doesn't only give access to the email you're waiting for but also all past and future emails, which can easily be abused, that's what I meant when saying that automating personal email addresses is sensitive, and trickier because email providers feature additional protection against this for that reason.

Please keep calm.

**Comment by Dhanamerdeka** [comment]
> The methods used to automatically receive and read emails doesn't only give access to the email you're waiting for but also all past and future emails, which can easily be abused, that's what I meant when saying that automating personal email addresses is sensitive, and trickier because email providers feature additional protection against this for that reason.

ahh i see...
i didn't bring that idea anyway... i think you're talking about my simple idea....
well... that's other idea had too many hole on it... 

that's why i suggesting to use some simple ways above...

**Comment by Pandapip1** [comment]
I still don't understand what's wrong with just bookmarking the HCaptcha accessibility URL. If all else fails, maybe the extension could force-whitelist that 3rd party cookie and automatically refresh it from a user-provided cookie URL?

I don't think it's any harder to sign up for the HCaptcha accessibility URL than it is to sign up to any of the speech recognition APIs. Considering the use-case of this extension is improving accessibility, I doubt the utility of trying to do anything more than providing a link to hCaptcha's accessibility sign up page and automatically refresh and force the cookie to load.

Maybe adding some code to implement privacy passes might also be useful, just to reduce the number of hCaptchas shown.

**Comment by Dhanamerdeka** [comment]
> I still don't understand what's wrong with just bookmarking the HCaptcha accessibility URL. If all else fails, maybe the extension could force-whitelist that 3rd party cookie and automatically refresh it from a user-provided cookie URL?
> 
> I don't think it's any harder to sign up for the HCaptcha accessibility URL than it is to sign up to any of the speech recognition APIs. Considering the use-case of this extension is improving accessibility, I doubt the utility of trying to do anything more than providing a link to hCaptcha's accessibility sign up page and automatically refresh and force the cookie to load.
> 
> Maybe adding some code to implement privacy passes might also be useful, just to reduce the number of hCaptchas shown.

well... if you answer like that.... then what's wrong doing some captcha work and solve it manually?

afterall... we on here want to everything to be simpler and automated.....
that's why we discussing in here.....

of all people just like you.... then no more automated things on this world.... as time goes by... we understand that time is the most expensive things on this world.... that's why something like this addons useful

and now you suggesting back to manual?
really dude...?

**Comment by dessant** [comment]
I have cleaned up the thread and kept the parts that are useful, please only share relevant information going forward.

I agree that automating the signup for the accessibility cookie is out of scope, the most this extension will do is to refresh the cookie if you have already signed up.

The main goal here is to find solutions that involve object recognition.

**Comment by Dhanamerdeka** [comment]
i hope the good idea can be implemented if they're possible.... :D

**Comment by Pandapip1** [comment]
> the most this extension will do is to refresh the cookie if you have already signed up.

I'd argue that it should probably implement https://github.com/privacypass/challenge-bypass-extension, if only to reduce the number of hCaptchas shown.

**Comment by kotx** [comment]
@Pandapip1 Usually, I find that Privacy Pass just doesn't work for hCaptcha :P. Also, this does not bypass the hCaptcha, which is the entire goal of this extension (bypassing captchas).

**Comment by Pandapip1** [comment]
It doesn't work because the privacy pass extension broke their code when
they added cloudflare and haven't realized it yet. There's an open PR in
their repo that fixes it, but it has yet to be merged. Also, the protocol
does bypass hCaptchas. Just not all of them :)

**Comment by Pandapip1** [comment]
Assuming the solver solves it 99% of the time, implementing privacy passes will improve that accuracy to 99.91%. I don't see any downside.

**Comment by kotx** [comment]
My mistake, I thought you meant the privacy pass solution would be the only thing the extension would offer. Sounds good now.

**Comment by Ezekiel-Game** [comment]
i dont know anything but i found this https://chrome.google.com/webstore/detail/hcaptcha-solver/lfpfbgeoodeejmjdlfjbfjkemjlblijg

**Comment by Ezekiel-Game** [comment]
> Ezekiel

it may be malicious i think

**Comment by leodoi3** [comment]
> i dont know anything but i found this https://chrome.google.com/webstore/detail/hcaptcha-solver/lfpfbgeoodeejmjdlfjbfjkemjlblijg

I tried it and there are several issues with it:
1. Not open source
2. Doesn't always work ; you don't know what's happening under the hood 
3. Redirects you to gtechmonitor affiliate link before reaching amazon

**Comment by Ezekiel-Game** [comment]
> > i dont know anything but i found this https://chrome.google.com/webstore/detail/hcaptcha-solver/lfpfbgeoodeejmjdlfjbfjkemjlblijg
> 
> I tried it and there are several issues with it:
> 
> 1. Not open source
> 2. Doesn't always work ; you don't know what's happening under the hood
> 3. Redirects you to gtechmonitor affiliate link before reaching amazon

i said it may be malicious did i

**Comment by Ezekiel-Game** [comment]
and i dont know anything so dont blame me :(

**Comment by chirag127** [comment]
> It doesn't work because the privacy pass extension broke their code when they added cloudflare and haven't realized it yet. There's an open PR in their repo that fixes it, but it has yet to be merged. Also, the protocol does bypass hCaptchas. Just not all of them :)
> […](#)
> Message ID: ***@***.***>

@ so I can install after merging,

**Comment by viasux** [comment]
> That requires a signup/email, though. It also requires login.

also it allows for them to constantly track you all over the internet

**Comment by CuteistCat** [comment]
A problem with image regonition is evry 2 weeks hcaptcha changes how the captchas work.

**Comment by CuteistFox** [comment]
Another idea is to have a system where you create a disposable email (on gmail or something to stop blacklisting) and use pop3 and scrape the emails but the cookies get shared on a server and every time a website has a captcha after you use the cookie it grabs a diffrent one from that server and tells it that the other one isnt in use to prevent tracking by hcaptcha.
You will probly need to make it so you need to have a email in the network to get a cookie so there isnt more users than cookies and a cookies shortage because you can only bake one cookie per email per day. (a simple client side check ofc)
